src-ui/src/app/components/common/confirm-dialog/confirm-dialog.component.ts

```diff
@@ -33,19 +33,21 @@ export class ConfirmDialogComponent {
 
   confirmButtonEnabled = true
   seconds = 0
+  secondsTotal = 0
 
   confirmSubject: Subject<boolean>
 
   delayConfirm(seconds: number) {
```

Comment:
I am not very happy with this implementation for the delay. If you can avoid setTimeout() and setIntervall() in Angular Applications you should. This will trigger the change detection for the whole app every time it triggers.

you can do this with rxjs as well.

something like:
```
const delay = 5000
const refreshIntervall = 100

return intervall(refreshIntervall).pipe(switchMap((count) => {
return {
  time : delay - (count * refreshIntervall), 
  buttonEnabled: (delay - (count * refreshIntervall) <= 0)}
}),
take(delay/refreshIntervall)
)
```

Reply:
Always appreciate your feedback @FrankStrieter . I rewrote with rxjs, albeit a little different than your suggestion. Welcome your thoughts, thanks